tez-api/src/main/java/org/apache/tez/dag/api/client/DAGClientImpl.java

```diff
@@ -408,6 +409,8 @@ private DAGStatus getDAGStatusViaAM(@Nullable Set<StatusGetOpts> statusOptions,
     } catch (ApplicationNotFoundException e) {
       LOG.info("DAG is no longer running - application not found by YARN", e);
       dagCompleted = true;
+    } catch (NoCurrentDAGException e) {
+      return dagLost(e);
```

Comment:
the exception `e` isn't used in the method ``dagLost``, we can remove that param., moreover I think we can explore having a info log with the exception before returning

Reply:
agreed